autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java

```diff
@@ -83,6 +95,10 @@ public ScalingAction govern(String serviceRef, ScalingAction action) {
                     if (lastInstanceInfo.getTotalInstances() < scalingConfiguration.getMinInstances()) {
                         return new ScalingAction(ScalingOperation.SCALE_UP,
                                                  scalingConfiguration.getMinInstances() - lastInstanceInfo.getTotalInstances());
+                    } else if (!shouldGovern(currentMemoryLimitStage, shutdownPriority)) {
```

Comment:
- Terminology is confusing (you're in a method called `govern` introducing a function called `shouldGovern`)
- Branch is duplicated (reorder to eliminate)
- I don't see why how this decision can be taken without reference to why are other service's minimum instances aren't met.